packages/contracts/src/dollar/libraries/LibUbiquityPool.sol

```diff
@@ -0,0 +1,345 @@
+                .unclaimedPoolCollateral[collateralAddress]
+                .sub(collateralAmount);
+
+            sendCollateral = true;
+
+            if (sendCollateral) {
+                IERC20(collateralAddress).transfer(
+                    msg.sender,
+                    collateralAmount
+                );
+            }
+        }
+    }
+
+    /// ADMIN FUNCTIONS ///
+
+    function addToken(
+        address collateralAddress,
+        IMetaPool collateralMetaPool
+    ) internal {
+        require(
+            collateralAddress != address(0x0) &&
+                address(collateralMetaPool) != address(0x0),
+            "0 address detected"
+        );
+        UbiquityPoolStorage storage poolStorage = ubiquityPoolStorage();
+        uint256 defaultDecimals = 18;
+        uint256 collateralDecimals = uint256(
+            IERC20Metadata(collateralAddress).decimals()
+        );
+
+        poolStorage.collateralAddresses.push(collateralAddress);
+        poolStorage.collateralMetaPools[collateralAddress] = collateralMetaPool;
+        poolStorage.missingDecimals[collateralAddress] = uint8(
+            defaultDecimals.sub(collateralDecimals)
+        );
+    }
+
+    function getRedeemCollateralBalances(
+        address account,
+        address collateralAddress
+    ) internal view returns (uint256) {
+        return
+            ubiquityPoolStorage().redeemCollateralBalances[account][
+                collateralAddress
+            ];
+    }
+
+    function setRedeemActive(
+        address collateralAddress,
+        bool redeemPaused_
+    ) internal {
+        ubiquityPoolStorage().collateralRedeemActive[
+            collateralAddress
+        ] = redeemPaused_;
+    }
+
+    function getRedeemActive(
+        address collateralAddress
+    ) internal view returns (bool) {
+        return ubiquityPoolStorage().collateralRedeemActive[collateralAddress];
+    }
+
+    function getMintActive(
+        address collateralAddress
+    ) internal view returns (bool) {
+        return ubiquityPoolStorage().collateralMintActive[collateralAddress];
+    }
+
+    function setMintActive(
+        address collateralAddress,
+        bool mintPaused_
+    ) internal {
+        ubiquityPoolStorage().collateralMintActive[
+            collateralAddress
+        ] = mintPaused_;
+    }
+
+    /// CHECK FUNCTIONS ///
+
+    function checkCollateralToken(
+        address collateralAddress
+    ) internal view returns (bool isCollateral) {
+        address[] memory collateralAddresses_ = ubiquityPoolStorage()
+            .collateralAddresses;
+        for (uint256 i; i < collateralAddresses_.length; ++i) {
+            if (collateralAddress == collateralAddresses_[i]) {
+                isCollateral = true;
+            }
+        }
+    }
+
+    /// CALC FUNCTIONS ///
+
+    function calcMintDollarAmount(
+        uint256 collateralAmountD18,
+        uint256 collateralPriceCurve3,
```

Comment:
What is `Curve3`? If it is referring to the Curve3Pool I think we should write it out like that!